fastapi/encoders.py

```diff
@@ -105,24 +105,22 @@ def jsonable_encoder(
                 )
             )
         return encoded_list
+    for encoder_type, encoder in custom_encoder.items():
+        if isinstance(obj, encoder_type):
+            return encoder(obj)
+    for encoder_type, encoder in ENCODERS_BY_TYPE.items():
+        if isinstance(obj, encoder_type):
+            return encoder(obj)
     errors: List[Exception] = []
     try:
-        if custom_encoder and type(obj) in custom_encoder:
-            encoder = custom_encoder[type(obj)]
-        else:
-            encoder = ENCODERS_BY_TYPE[type(obj)]
-        return encoder(obj)
-    except KeyError as e:
+        data = dict(obj)
+    except Exception as e:
         errors.append(e)
         try:
-            data = dict(obj)
+            data = vars(obj)
```

Comment:
Can I ask what is the rationale for using `vars` here instead of `dict`?

Reply:
I retained the previous behaviour but since I am using explicit instance checking now instead of catching a keyerror all the exception catching was moved 1 indentation to the left. It is the github diff displaying that makes it looks as if I changed this.